### PR TITLE
Cli bug fixes

### PR DESCRIPTION
**Pull request by Naman-Monga**

### CODE REVIEW

File: cli/oyster-cvm/src/main.rs [hunk 1]
```diff
@@ -34,12 +34,16 @@ enum Commands {
         docker_compose: String,
 
         /// List of Docker image .tar file paths
-        #[arg(short = 'i', long, required = true)]
-        docker_images: Vec<String>,
+        #[arg(short = 'i', long, required = false)]
```
Comment (roshanrags) [inline cli/oyster-cvm/src/main.rs:37]: can just set `default_value=""` equivalent, avoids Option everywhere

**Review by roshanrags — APPROVED** [review]

File: cli/oyster-cvm/src/main.rs [hunk 2]
```diff
@@ -36,12 +36,20 @@ enum Commands {
...
 
         /// Output folder name
         #[arg(short, long, default_value = "result")]
         output: String,
+
+        /// Git commit reference for oyster-monorepo
+        #[arg(
+            short = 'r',
+            long,
+            default_value = "3e6dbc844b42281462e65d7742d9436d4205fcfd"
```
Comment (vg-27) [inline cli/oyster-cvm/src/main.rs:50]: why is this exact commit used as default? need comment to instruct if this needs to be changed in future. When does a commit qualify for a default value here.

File: cli/oyster-cvm/src/commands/build.rs [hunk 3]
```diff
@@ -1,46 +1,97 @@
...
+        let line = line_result.context("Failed to read line from stdout")?;
+        info!(target: "nix", "{}", line);
+    }
+
+    info!("Build finished");
+
+    Ok(())
+}
+
+fn check_trusted_user() -> Result<()> {
```
Comment (vg-27) [inline cli/oyster-cvm/src/commands/build.rs:65]: this function is repeated from doctor, can be included in common util

File: cli/oyster-cvm/src/commands/build.rs [hunk 4]
```diff
@@ -1,46 +1,97 @@
...
 
 pub fn build_oyster_image(
     platform: Platform,
     docker_compose: &str,
     docker_images: &[String],
     output: &str,
+    commit_ref: &str,
 ) -> Result<()> {
+    info!("Checking if the current user is in the nix trusted-users list");
+    check_trusted_user()?;
```
Comment (vg-27) [inline cli/oyster-cvm/src/commands/build.rs:15]: why are we using this check here? what happens with nix command if this is not checked?

**Review by vg-27 — APPROVED** [review]

File: cli/oyster-cvm/src/main.rs (hunk 2 above)
Comment (Naman-Monga) [reply cli/oyster-cvm/src/main.rs:50]: any commit with the current nix configs will be relevant. Will have to be changed incase we make any changes to nix configs (add any param / change the project structure).
Will add a comment in the next pr.

File: cli/oyster-cvm/src/commands/build.rs (hunk 4 above)
Comment (Naman-Monga) [reply cli/oyster-cvm/src/commands/build.rs:15]: Actually this was was added as the error wasn't readable if trusted user was failing, but with nix user would instantly know. 
But yeah, to make things more organized, we can add a dependency checker, which scans for all things necessary in the env for that specific command.

File: cli/oyster-cvm/src/main.rs (hunk 2 above)
Comment (Naman-Monga) [reply cli/oyster-cvm/src/main.rs:50]: done

File: cli/oyster-cvm/src/commands/build.rs (hunk 4 above)
Comment (Naman-Monga) [reply cli/oyster-cvm/src/commands/build.rs:15]: removed from build, this will only run with doctor.

**Pull request merged**